Main.py

```diff
@@ -29,21 +29,36 @@ def run():
         still_in_cycle = 1 # 1
         while still_in_cycle:
 
-            t += MINIMUM_TIME_RESOLUTION # Possible speed up? find biggest time increment...
-
             # Check lag
             if lag:
+                time_step = MINIMUM_TIME_RESOLUTION
                 lagging = (sim_env['lag_time'][:nr_alive] > 0)
                 lag = any(lagging)
-                sim_env['lag_time'][lagging] -= MINIMUM_TIME_RESOLUTION
-                sim_env['next_divition'][lagging] += MINIMUM_TIME_RESOLUTION
+                sim_env['lag_time'][lagging] -= time_step
+                sim_env['next_divition'][lagging] += time_step
+            else:
+                if time_step != MINIMUM_TIME_RESOLUTION or np.random.random() < A_MYSTICAL_SPEED_UP_PARAMETER:
+                        soonest_division = np.min([sim_env['next_divition'][:nr_alive]])
+                        time_step = np.max([MINIMUM_TIME_RESOLUTION, soonest_division - t]) # find biggest time increment
```

Comment:
shouldn't this be `min`? Or maybe you handle it below, but it feels most rational (possibly not fastest though), to either check next time someone needs to divide or next time we need to record the population size.

Another way to go about it to only check once every 20 minutes and then fix everything that should have divided since last time.

Lets assume time `t` somewhere in the middle of the experiment.

Since the progression when in a cell cycle is linear and non-stochastic, the dividing cell (`i`) and its offspring (`o`) can have their cell cycle times and and progressions recalculated for `t` as long as the update frequency is much less than the shortest possible generation time of a cell reconstruction of the population at `t` should be possible from the `t-1` state and knowledge of when between `t-1` and `t` that the cells divided. 

If sampling-frequency is required to be less than .5 of expected minimum generation times, this optimization should be possible and ought to factor up speed by many magnitudes if I understand correctly how it is done now.